hibernate-core/src/main/java/org/hibernate/engine/jdbc/env/internal/JdbcEnvironmentImpl.java

```diff
@@ -84,8 +83,10 @@ public JdbcEnvironmentImpl(ServiceRegistryImplementor serviceRegistry, Dialect d
 		identifierHelperBuilder.setNameQualifierSupport( nameQualifierSupport );
 
 		IdentifierHelper identifierHelper = null;
+		ExtractedDatabaseMetaDataImpl.Builder dbMetaDataBuilder = new ExtractedDatabaseMetaDataImpl.Builder( this );
 		try {
 			identifierHelper = dialect.buildIdentifierHelper( identifierHelperBuilder, null );
```

Comment:
Yes, that's correct. In the case that JDBC DatabaseMetadata is not available and the DB really does support named parameters, the dialect can be overridden so that `#supportsNamedParameters` returns `true`.

What is the case that you think it will work incorrectly?